internal/slicer/slicer.go

```diff
@@ -397,3 +434,148 @@ func createFile(targetPath string, pathInfo setup.PathInfo) (*fsutil.Entry, erro
 		MakeParents: true,
 	})
 }
+
+type generateManifestsOptions struct {
+	packageInfo []*archive.PackageInfo
+	selection   []*setup.Slice
+	report      *Report
+	targetDir   string
+}
+
+func generateManifests(options *generateManifestsOptions) error {
+	manifestSlices, err := locateManifestSlices(options.selection)
```

Comment:
Shouldn't this function be present and somewhat tested under the manifest package instead?  The content is heavily dependent on the format defined there, and it would be nice to have tests for all these fields which don't look like entire integration tests, to make sure we're actually getting these details right without the massive body of tests which exist for the slicer.

Reply:
Done in https://github.com/canonical/chisel/pull/142/commits/9b03b95d4550f7fdfb8a319ba8b0faa1d3a885e1.

Reply:
This wasn't about just locateManifestSlices, but about the generation of the manifest itself.  All input parameters here come from out of the slicer, and all the behavior inside this function is about manifest generation. It seems misplaced at the slicer package.  That's what the comment above was about, and why I was talking about testing, etc. Sorry for not being clear.